Error building libmpv on Mac

I'm trying to build with these mpv_options:

```
-Dlibmpv=true
-Dlua=disabled
-Djavascript=disabled
-Djavascript=disabled
-Dvapoursynth=disabled
-Dswift-build=disabled
```

I get this error:

```
[224/229] Compiling Objective-C object libmpv.2.dylib.p/osdep_mac_app_bridge.m.o
FAILED: libmpv.2.dylib.p/osdep_mac_app_bridge.m.o 
clang -Ilibmpv.2.dylib.p -I. -I.. -Icommon -Ietc -Iplayer/javascript -Iplayer/lua -Isub -ITOOLS/osxbundle -I/Users/flavio/projects/mpv-build/build_libs/include -I/usr/local/Cellar/fontconfig/2.14.2/include -I/usr/local/Cellar/harfbuzz/8.1.1/include/harfbuzz -I/usr/local/Cellar/glib/2.76.4/include/glib-2.0 -I/usr/local/Cellar/glib/2.76.4/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I/usr/local/Cellar/pcre2/10.42/include -I/usr/local/Cellar/graphite2/1.3.14/include -I/usr/local/Cellar/fribidi/1.0.13/include/fribidi -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/opt/libpng/include/libpng16 -I/usr/local/Cellar/little-cms2/2.15/include -I/usr/local/Cellar/jpeg-turbo/3.0.0/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -O3 -D_FILE_OFFSET_BITS=64 -Wdisabled-optimization -Wempty-body -Wformat -Wimplicit-fallthrough -Wparentheses -Wpointer-arith -Wshadow -Wundef -Wvla -Wno-cast-function-type -Wno-format-zero-length -Wno-missing-field-initializers -Wno-sign-compare -Wno-switch -Wno-unused-parameter -fno-math-errno -fno-signed-zeros -fno-trapping-math -Werror=format-security -D_GNU_SOURCE -Wmissing-prototypes -Wstrict-prototypes -Werror=implicit-function-declaration -Wno-pointer-sign -DPL_HAVE_PTHREAD -DPL_STATIC -MD -MQ libmpv.2.dylib.p/osdep_mac_app_bridge.m.o -MF libmpv.2.dylib.p/osdep_mac_app_bridge.m.o.d -o libmpv.2.dylib.p/osdep_mac_app_bridge.m.o -c ../osdep/mac/app_bridge.m
../osdep/mac/app_bridge.m:115:7: error: use of undeclared identifier 'AppHub'
    [[AppHub shared] startRemote];
```

Looks related to having disabled "swift-build".

If I remove `-Dswift-build=disabled` I get:

```
FAILED: osdep/mac/swift.swiftmodule osdep/mac/swift.h osdep/mac/swift.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -enable-objc-interop -emit-objc-header -parse-as-library -O -D HAVE_MACOS_11_FEATURES -D HAVE_MACOS_COCOA_CB -module-name swift -emit-module-path osdep/mac/swift.swiftmodule -import-objc-header /Users/flavio/projects/mpv-build/mpv/osdep/mac/app_bridge_objc.h -emit-objc-header-path osdep/mac/swift.h -o osdep/mac/swift.o ../osdep/mac/application.swift ../osdep/mac/app_hub.swift ../osdep/mac/event_helper.swift ../osdep/mac/input_helper.swift ../osdep/mac/libmpv_helper.swift ../osdep/mac/log_helper.swift ../osdep/mac/menu_bar.swift ../osdep/mac/option_helper.swift ../osdep/mac/precise_timer.swift ../osdep/mac/presentation.swift ../osdep/mac/swift_compat.swift ../osdep/mac/swift_extensions.swift ../osdep/mac/type_helper.swift ../video/out/mac/common.swift ../video/out/mac/title_bar.swift ../video/out/mac/view.swift ../video/out/mac/window.swift ../video/out/cocoa_cb_common.swift ../video/out/mac/gl_layer.swift -I. -I/Users/flavio/projects/mpv-build/mpv -I/Users/flavio/projects/mpv-build/build_libs/include -I/Users/flavio/projects/mpv-build/build_libs/include
../osdep/mac/libmpv_helper.swift:134:20: warning: 'CGLFlushDrawable' was deprecated in macOS 10.14: OpenGL API deprecated. (Define GL_SILENCE_DEPRECATION to silence these warnings)
        if !skip { CGLFlushDrawable(ctx) }
                   ^
../osdep/mac/swift_compat.swift:50:42: error: 'kCGColorSpaceITUR_2100_HLG' has been renamed to 'CGColorSpace.itur_2100_HLG'
    static let itur_2100_HLG: CFString = kCGColorSpaceITUR_2100_HLG
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
                                         CGColorSpace.itur_2100_HLG
CoreGraphics.kCGColorSpaceITUR_2100_HLG:3:12: note: 'kCGColorSpaceITUR_2100_HLG' was obsoleted in Swift 3
public let kCGColorSpaceITUR_2100_HLG: CFString
           ^
../osdep/mac/swift_compat.swift:51:41: error: 'kCGColorSpaceITUR_2100_PQ' has been renamed to 'CGColorSpace.itur_2100_PQ'
    static let itur_2100_PQ: CFString = kCGColorSpaceITUR_2100_PQ
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~
                                        CGColorSpace.itur_2100_PQ
```

--- a/meson.build
+++ b/meson.build
@@ -1540,6 +1540,10 @@ swift = get_option('swift-build').require(
 )
 features += {'swift': swift.allowed()}
 
+if features['cocoa'] and not features['swift'] and get_option('cplayer')
+    error('cplayer enabled but no suitable swift version could be found or was disabled!')
+endif
+
 swift_sources = []
 if features['cocoa'] and features['swift']
     swift_sources += files('osdep/mac/application.swift',

--- a/osdep/mac/app_bridge.m
+++ b/osdep/mac/app_bridge.m
@@ -110,6 +110,7 @@ const struct m_sub_options *app_bridge_vo_conf(void)
     return &vo_sub_opts;
 }
 
+#if HAVE_SWIFT
 void cocoa_init_media_keys(void)
 {
     [[AppHub shared] startRemote];
@@ -139,4 +140,4 @@ int cocoa_main(int argc, char *argv[])
 {
     return [(Application *)[Application sharedApplication] main:argc :argv];
 }
-
+#endif

